privacyidea/lib/eventhandler/webhookevanthandler.py

```diff
@@ -0,0 +1,150 @@
+import json
+import logging
+import requests
+from requests.exceptions import HTTPError, Timeout, ConnectionError, RequestException
+from collections import defaultdict
+
+log = logging.getLogger(__name__)
+
+
+class CONTENT_TYPE(object):
+    """
+    Allowed type off content
+    """
+    JSON = "json"
+    URLENCODED = "urlendcode"
+
+
+class ACTION_TYPE(object):
+    """
+    Allowed actions
+    """
+    POST_WEBHOOK = "post_webhook"
+
+
+class WebHookHandler(BaseEventHandler):
+    """
+    With the WebHook Handler, a webhook can be sent at a particular event
+    """
+
+    identifier = "WebHookHandler"
+    description = "This eventhandler can post webhooks"
+
+    @property
+    def allowed_positions(cls):
+        """
+        This returns the allowed positions of the event handler definition.
+
+        :return: list of allowed positions
+        """
+        return ["post", "pre"]
+
+    @property
+    def actions(cls):
+        """
+        This method returns a dictionary of allowed actions and possible
+        options in this handler module.
+
+        :return: dict with actions
+        """
+        # The event handler has just one action. Maybe we can  hide action select for the clarity of the UI
+        actions = {ACTION_TYPE.POST_WEBHOOK: {
+            "URL": {
+                "type": "str",
+                "required": True,
+                "description": _("The URL the WebHook is posted to")
+            },
+            "content_type": {
+                "type": "str",
+                "required": True,
+                "description": _("The encoding that is sent to the WebHook, for example json"),
+                "value": [
+                    CONTENT_TYPE.JSON,
+                    CONTENT_TYPE.URLENCODED]
+            },
+            "data": {
+                "type": "str",
+                "required": True,
+                "description": _("The data posted in the WebHook")
+            }
+        }}
+        return actions
+
+    def do(self, action, options=None):
+        """
+        This method executes the defined action in the given event.
+
+        :param action: The action to perform
+        :type action: str
+        :param options: Contains the flask parameters g, URL, data, parameter and the fire and forget
+        :type options: dict
+        :return:
+        """
+        ret = True
+        g = options.get("g")
+        request = options.get("request")
+        handler_def = options.get("handler_def")
+        handler_options = handler_def.get("options")
+        webhook_url = handler_options.get("URL")
+        webhook_text = handler_options.get("data")
+        content_type = handler_options.get("content_type")
+
+        # Replace placeholder in data
+        logged_in_user = g.logged_in_user.get("username")
+        attributes = {
+            'logged_in_user': logged_in_user
+        }
+        webhook_text = webhook_text.format_map(defaultdict(str, attributes))
```

Comment:
I can not reproduce, that this works. Why not do:

    webhook_text = webhook_text.format(**attributes)

Where does the 'str' come from?
How does the json, you would be allowed to use in the `data` work with the python formatting?

Reply:
The problem with form(**attributes) is if there is a placeholder in the string and in attributes is no matching key, the programme will crash. For example, I have the string a = 'This will crash because {resonen}' and the 
attributes = {"problem": "1","mistake": "2"}, it will crash because he can't found the key 'resonen'.

Reply:
https://github.com/privacyidea/privacyidea/pull/3154/commits/63296154129247c5d0cf0c73c7119668bd1a02f0